internal/rickandmorty/character_interactor.go

```diff
@@ -79,3 +83,79 @@ func (i interactor) GetCharactersStoredByPageFromAPI(page int, storageFormat str
 	err := i.storeDS.Read(filename, &characterList, storageFormat)
 	return characterList, err
 }
+
+type lineChecker func(number int) bool
+
+func isEven(number int) bool {
+	return number%2 == 0
+}
+
+func isOdd(number int) bool {
+	return !isEven(number)
+}
+
+func worker(ctx context.Context, id int, processed chan<- Character, lines <-chan string, params ReadConcurrentParams) {
+	codec := NewCSVCharacterCodec()
+	count := 0
+	var isValidLine lineChecker = isEven
+	if params.Type == "odd" {
+		isValidLine = isOdd
+	}
+loop:
+	for line := range lines {
+		character := Character{}
```

Comment:
If line 106 is inside the for loop, every single time you are creating a new variable; and worst-case scenario, you don't even use it.

By moving it out of the loop you already have a variable initialized, and ready to be used inside the loop.
```suggestion
	character := Character{}
	for line := range lines {
```